docs/jwst.rst

```diff
@@ -0,0 +1,254 @@
+or equivalently the image on the detector is moved closer to the OTE exit pupil
+than best focus. 
+
+Note that the weak lenses are in the short wave channel only; webbpsf won't stop
+you from simulating a LW image with a weak lens, but that's not a real
+configuration that can be acheived with NIRCam.
+
+
+SI WFE
+------
+
+(Not yet available)
+
+The SI internal WFE measurements are distinct for each of the modules and
+channels. When enabled, these are added to the final pupil of the optical
+train, i.e. after the coronagraphic image planes. 
+
+
+NIRSpec
+=======
+
+Imaging and spectroscopy
+------------------------
+
+webbpsf models the optics of NIRSpec, mostly in **imaging** mode or for monochromatic PSFs that can be assembled into spectra using other tools.
+
+This is not a substitute for a spectrograph model, but rather a way of
+simulating a PSF as it would appear with NIRSpec in imaging mode (e.g. for
+target acquisition).  It can also be used to produce monochromatic PSFs
+appropriate for spectroscopic modes, but other software must be used for
+assembling those monochromatic PSFs into a spectrum.
+
+Slits: webbpsf includes models of each of the fixed slits in NIRSpec (S200A1, S1600A1, and so forth), plus a
+few patterns with the MSA: (1) a single open shutter, (2) three adjacent
+open shutters to make a mini-slit, and (3) all shutters open at once.
+Other MSA patterns could be added if requested by users. 
+
+By default the ``pupil_mask`` is set to the "NIRSpec grating" pupil mask.  In
+this case a rectangular pupil mask 8.41x7.91 m as projected onto the primary is
+added to the optical system at the pupil plane after the MSA. This is an
+estimate of the pupil stop imposed by the outer edge of the grating clear
+aperture, estimated based on optical modeling by Erin Elliot and Marshall
+Perrin.
+
+
+SI WFE
+------
+
+(Not yet available)
+
+SI WFE will most likely be added to the entrance pupil, prior to the MSA image plane. This model is still under development. 
+
+NIRISS
+======
+
+
+Imaging and AMI
+----------------
+
+WebbPSF models the direct imaging and nonredundant aperture masking interferometry modes of NIRISS in the usual manner.
+
+Note that long wavelength filters (>2.5 microns) are used with a pupil
+obscuration which includes the pupil alignment reference fixture. This is called
+the "CLEARP" pupil.
+
+Based on the selected filter, webbpsf will automatically toggle the
+``pupil_mask`` between "CLEARP" and the regular clear pupil (i.e.
+``pupil_mask=None``).
+
+
+Slitless Spectroscopy
+---------------------
+
+webbpsf provides preliminary support for 
+the single-object slitless
+spectroscopy ("SOSS") mode using the GR700XD cross-dispersed grating. Currently
+this includes the clipping of the pupil due to the undersized grating and its
+mounting hardware, and the cylindrical lens that partially defocuses the light
+in one direction.
+
+.. warning ::
```

Comment:
Based on GitHub's syntax highlighting, I think there shouldn't be a space after `warning` and before `::`.